### PR TITLE
Group pagination

### PR DESCRIPTION
**Pull request by schreiaj**
patient-pagination

### CODE REVIEW

File: app/controllers/patients/index.js [hunk 1]
```diff
@@ -75,6 +56,19 @@ export default Controller.extend({
...
   }),
 
+  groupParams: computed('groupId', {
+    get() {
+      let groupId = this.get('groupId');
+      return groupId ? { '_query': 'group', groupId } : {};
+    }
+  }),
+
+  queryParams: computed('groupParams', {
```
Comment (noranda) [inline app/controllers/patients/index.js]: `queryParams` is a reserved key in Ember

File: app/controllers/patients/index.js [hunk 2]
```diff
@@ -42,28 +39,12 @@ export default Controller.extend({
...
     }
   }),
 
-  sortedPatients: computed('selectedPopulations.length', 'filteredPatients.@each.fullName', function sortedPatients() {
-    if (this.get('selectedPopulations.length') === 0) {
-      return this.get('filteredPatients');
-    }
-
-    return this.get('filteredPatients').sortBy('fullName');
+  sortedPatients: computed('filteredPatients.@each.fullName', function sortedPatients() {
```
Comment (noranda) [inline app/controllers/patients/index.js]: Change this to: 

``` js
sortedPatients: computed.reads('filteredPatients')
```

You're using a computed property on a key that you don't do anything with, it's better to use `computed.reads` if you're just making an alias.

File: app/controllers/patients/index.js [hunk 3]
```diff
@@ -42,28 +39,12 @@ export default Controller.extend({
 
   populationPatients: computed('selectedPopulations.[]', {
```
Comment (noranda) [inline app/controllers/patients/index.js]: Change this to:

``` js
populationPatients: computed.reads('model.patients')
```

Same reason as the change to `sortedPatients`

File: app/controllers/patients/index.js [hunk 4]
```diff
@@ -17,14 +18,10 @@ export default Controller.extend({
   riskLowValue: 1,
   riskHighValue: 4,
   interventionTypes: [],
+  groupId: '',
 
   totalPatients: computed('model.patients.meta.total', 'selectedPopulations.length', 'populationPatients.length', function totalPatients() {
```
Comment (noranda) [inline app/controllers/patients/index.js]: Change this to:

``` js
totalPatients: computed.reads('model.patients.meta.total')
```

File: app/controllers/patients/index.js [hunk 5]
```diff
@@ -4,9 +4,10 @@ import run from 'ember-runloop';
 import { A } from 'ember-array/utils';
 
 export default Controller.extend({
-  queryParams: ['page', { currentAssessment: 'risk_assessment' }, 'sortBy', 'sortDescending'],
+  // queryParams: ['page', { currentAssessment: 'risk_assessment' },  'sortBy', 'sortDescending', 'groupId'],
```
Comment (noranda) [inline app/controllers/patients/index.js:7]: Why was this removed?

File: app/utils/fhir-paged-remote-array.js [hunk 6]
```diff
@@ -4,9 +4,8 @@ import computed from 'ember-computed';
 export default PagedRemoteArray.extend({
   page: 1,
   totalPages: 0,
-
   sortBy: null,
-  sortDescending: null,
+  sortDesc: false,
```
Comment (noranda) [inline app/utils/fhir-paged-remote-array.js]: Why was this changed? Line 17 of this file still reads from `sortDescending` which you removed here.

**Comment by noranda** [comment]
What branch of the ie server does this require? When I try to filter the patient list by population I get an error from the go server:

screen shot 2016-03-21 at 4 09 51 pm

**Comment by cmoesel** [comment]
Turns out we never merged the necessary changes to IE.  For now, use ie's "group_resolution_hack" branch to test it.

**Comment by noranda** [comment]
@cmoesel: Tested on the `group_resolution_hack` branch with the same exception on the go server. Can you verify this works with the most recent changes?

**Comment by cmoesel** [comment]
@noranda: Doh.  I think I might know what is going on.  That branch of ie will _not_ work with the HEAD of fhir master-- due to changes that were just merged into fhir today.  You could try reverting fhir to a previous commit (`git checkout -b old_fhir 42311fd78d29453dc5c62d115e089aedc64887a5`) -- or you can wait for me to rebase 'group_resolution_hack' on the new ie code to get it up to date with fhir.  I can do that tomorrow morning.

Sorry for the trouble.

**Comment by noranda** [comment]
@cmoesel: No problem. I will wait for the rebase!

**Comment by cmoesel** [comment]
@noranda: Rebase is complete.  That said, since it is a rebase, you can't do a normal pull on the branch (because I rewrote history), so you'll want to do this instead (in your ie repo):

```
git checkout group_resolution_hack
git fetch
git reset --hard origin/group_resolution_hack
```

That will completely replace your local group_resolution_hack branch with the remote one.

Another thing to note -- we now use Go vendoring, which is a feature of Go 1.6.  Ideally, you should update your Go to v1.6.  If you want to stay on Go 1.5 for some reason, then you need to set an environment variable before you do the `go build` or `go run server.go`:

```
# NOTE: Only needed for Go 1.5.  Vendoring is on by default in Go 1.6
export GO15VENDOREXPERIMENT=1
```

**Comment by cmoesel** [comment]
@schreiaj and @noranda: When I tested this, I ran into one bug.  Here is how to reproduce:
1. Go to patients page
2. Click on page 2
3. Select a population (group)
4. Unselect the population (group)

After step 4, it doesn't go back to the full list of patients.  It's stuck.
